token-service/src/main/java/org/p2p/token/service/repository/metadata/TokenMetadataRemoteRepository.kt

```diff
@@ -38,14 +38,15 @@ internal class TokenMetadataRemoteRepository(
     ): UpdateTokenMetadataResult = withContext(dispatchers.io) {
         val response = try {
             api.getZipFile("${tokenServiceUrl}get_all_tokens_info", ifModifiedSince)
-        } catch (e: HttpException) {
-            return@withContext if (e.code() == HttpURLConnection.HTTP_NOT_MODIFIED) {
-                UpdateTokenMetadataResult.NoUpdate
-            } else {
-                UpdateTokenMetadataResult.Error(e)
-            }
         } catch (e: Throwable) {
-            return@withContext UpdateTokenMetadataResult.Error(e)
+            return@withContext when (e) {
```

Comment:
to a method

Reply:
and this is very hard to read 
try doin `if (e is HttpException && e.code() == HttpURLConnection.HTTP_NOT_MODIFIED) { NoUpdate } else  UpdateTokenMetadataResult.Error

Reply:
zaeba =)

Reply:
`fixed`